frame/contracts/common/Cargo.toml

```diff
@@ -0,0 +1,19 @@
+[package]
+name = "pallet-contracts-common"
```

Comment:
Yeah that was a tough one.

We have all our primitives living in `/primitives` but if we share some piece of code they live in a common package. Or used to. Now, it seems that we do not use this convention anymore.

Do you think I should move it to `/primitives` as well? Otherwise it would be kind of weird that it is kind of primitives but not in primitives dir?

Reply:
It seems we only have `consensus-common` and `executor-common`, The first one seems like a good name for me, cause it's the common part between multiple consensus algorithms. `primitive` to me seems more like a thing that is defining the interface/interoperation between two separate modules. And I think this is exactly how that crate is being used.

However I don't think it should be moved to `/primitives`, that directory is special  as it's the kind of primitive types that are defining (currently virtual) repository/logical boundaries between `/frame`, `/client` and `node`. The RPC is a bit tricky, cause it will always depend on `/frame` and `/primitives`, I'm actually totally fine with keeping it within `/frame`, hence the `contract-primitives` could stay in `/frame` as well.

Reply:
Thanks for the explaintations, really helpful!